src/libcore/iter/traits/iterator.rs

```diff
@@ -1684,6 +1684,7 @@ pub trait Iterator {
     /// assert_eq!(it.len(), 2);
     /// assert_eq!(it.next(), Some(&40));
     /// ```
+    #[allow(clippy::while_let_on_iterator)] // uses mutable borrow resulting in worse assembly
```

Comment:
I don't think we should add this to libstd, at least for now; I'm not sure libs team wants these scattered around.

Reply:
Sure I'll remove all Clippy annotations.
One day if Clippy is expected to be over Rust codebase this will be probably unavoidable though. Probably topic for a meeting.

Reply:
Also dropped https://github.com/mati865/rust/commit/694e62518ad135cd3a6a2f139528e70773f4b849